addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -1302,11 +1335,136 @@ var VideoWidget = MediaWidget.extend({
             '</div>'
         );
     },
+    /**
+     * Get a video element wrapped for the website builder. Used for uploaded videos.
+     *
+     * @param video The video element.
+     */
+    getWrappedVideo: function (video) {
+        return $(
+            '<div class="media_iframe_video" data-oe-expression="' + video.attr('src') + '">' +
```

Comment:
don't create html by concatanating strings, this is a recipe for xss

Reply:
> Hello @kzhekov nice to see this is moving forward.
> 
> Here is a first (very) superficial review, made when reading the PR transversely to get a first idea of the code.
> 
> Most remarks are to be applied on the whole diff, I didn't comment everywhere, that would have been a bit long yum
> 
> Note that I only had a single .mp4 file inside my download folder, not sure where it exactly came from, probably from an odoo task/opw, but it doesn't let me upload it. It is not recognized as a mp4 video. I then downloaded a mp4 video sample and tried again, still not letting me go through. Here is the files [File 1](https://drive.google.com/file/d/1cgqC1T5EcYS-2_olQ7OgXUtVYmc39nAU/view?usp=sharing) and [file 2](https://drive.google.com/file/d/1rePn38qOhwnkPtY_8PZUGq63I87IQ9B5/view?usp=sharing)

Thank you for the review! I've addressed most of your comments and resolved those where the changes were pushed.

I've updated the binary signatures to include the two file types that failed. Unfortunately, mp4 is a bundle format and it does regroup a lot of different signatures. I didn't find any reliable sources that listed them, so I had to build the list of signatures through trial and error 😞